src/main/scala/org/broadinstitute/hail/driver/SampleQC.scala

```diff
@@ -253,9 +251,18 @@ object SampleQC extends Command {
         vds
           .rdd
           .treeAggregate(Array.fill[SampleQCCombiner](vds.nSamples)(new SampleQCCombiner))({ case (acc, (v, (va, gs))) =>
-            val vIsSingleton = gs.iterator.existsExactly1(_.isCalledNonRef)
+
+            val ACs = gs.foldLeft(Array.fill(v.nAltAlleles)(0))({
+              case (acc, g) =>
+                g.gt.map({
+                  call =>
+                    if (call > 0)
+                      Genotype.gtPair(call).alleleIndices.filter(_ > 0).foreach(x => acc(x - 1) += 1)
+                })
```

Comment:
I would rewrite this as:

```scala
g.gt
  .filter(_ > 0)
  .foreach(call => Genotype.getPair(call).alleleIndices.filter(_ > 0).foreach(x => acc(x - 1) += 1)
```

At the very least, `foreach` should be used since the lambda is effectful (i.e. non-pure) and has no return value.